packages/neuron-wallet/src/services/addresses.ts

```diff
@@ -27,14 +27,33 @@ export default class AddressService {
   private static minUnusedAddressCount: number = 3
 
   private static async create(addresses: AddressInterface[]) {
-    const publicKeyInfos = addresses.map(addr => {
-      return HdPublicKeyInfo.fromObject({
-        ...addr,
-        publicKeyInBlake160: addr.blake160
+    const walletIds = addresses.map(v => v.walletId)
+    const walletIdMaxAddressIndex = await getConnection()
+      .getRepository(HdPublicKeyInfo)
+      .createQueryBuilder()
+      .select('walletId')
+      .addSelect('group_concat(publicKeyInBlake160)', 'publicKeyInBlake160Array')
+      .where({ walletId: In(walletIds) })
```

Comment:
This could be simplified to `walletId = walletId` since all addresses belong to the same wallet.

To ensure this, we can add an assert in this method.